### PR TITLE
feat(cloud-monitoring-exporter): Support Distribution exporting

### LINKED ISSUE #101: Add distribution and histogram metric support
Only non-distribution (bool, int64, double, etc.) values are supported by the exporter at the moment. See [this](https://github.com/GoogleCloudPlatform/opentelemetry-operations-js/blob/a1f6264bec877bc00c66348d249ac076d8cf0207/packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts#L177) line of code for the relevant TODO and where support should be added (perhaps among other places).

### PR DESCRIPTION
**Pull request by markrzen**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue/feature](https://github.com/GoogleCloudPlatform/opentelemetry-operations-js/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Fixes #101

### CODE REVIEW

File: packages/opentelemetry-cloud-monitoring-exporter/package.json [hunk 1]
```diff
@@ -44,19 +44,19 @@
...
+    "mocha": "^10.0.0",
+    "nock": "^13.2.9",
     "nyc": "15.1.0",
     "rimraf": "3.0.2",
-    "sinon": "12.0.1",
-    "ts-mocha": "9.0.2",
-    "typescript": "4.5.4"
+    "sinon": "^14.0.0",
+    "ts-mocha": "^10.0.0",
+    "typescript": "^4.8.3"
```
Comment (markrzen) [inline packages/opentelemetry-cloud-monitoring-exporter/package.json:59]: Updating all the dev deps related to testing as house-keeping.

File: packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts [hunk 2]
```diff
@@ -73,20 +76,27 @@ function transformMetricKind(kind: OTMetricKind): MetricKind {
   switch (kind) {
     case OTMetricKind.COUNTER:
     case OTMetricKind.OBSERVABLE_COUNTER:
+    case OTMetricKind.HISTOGRAM:
       return MetricKind.CUMULATIVE;
```
Comment (markrzen) [inline packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts:80]: Histogram when converted to a GCM Distribution is apparently best represented by MetricKind.CUMULATIVE, as MetricKind.DELTA is not allowed for Custom Metrics.

File: packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts [hunk 3]
```diff
@@ -73,20 +76,27 @@ function transformMetricKind(kind: OTMetricKind): MetricKind {
...
-function transformValueType(valueType: OTValueType): ValueType {
+function transformValueType(
+  valueType: OTValueType,
+  metricKind: OTMetricKind
+): ValueType {
+  // Stackdriver ValueType is not a 1 to 1 with OTel ValueType
   if (valueType === OTValueType.DOUBLE) {
+    if (metricKind === OTMetricKind.HISTOGRAM) {
+      return ValueType.DISTRIBUTION;
+    }
```
Comment (markrzen) [inline packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts:99]: Histograms are both a OTValueType.DOUBLE and a OTMetricKind.HISTOGRAM.

File: packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts [hunk 4]
```diff
@@ -144,6 +157,7 @@ function transformPoint(
   switch (metricDescriptor.metricKind) {
     case OTMetricKind.COUNTER:
     case OTMetricKind.OBSERVABLE_COUNTER:
+    case OTMetricKind.HISTOGRAM:
```
Comment (markrzen) [inline packages/opentelemetry-cloud-monitoring-exporter/src/transform.ts:160]: Histograms require a startDate.

**Comment by markrzen** [comment]
Working on the CLA, will be a few days at least.

**Comment by markrzen** [comment]
@dashpole Can you confirm this repository is still being supported for contributions?

**Comment by dashpole** [comment]
Please make sure you sign the CLA, thanks!

**Comment by markrzen** [comment]
@dashpole Fantastic that the project is still alive. I am going through the corp cla process, so it may take a few days. Once this PR is ready and merged, I may put in a subsequent one to upgrade to the latest Otel libraries, unless someone is already working on that.

**Pull request closed without merging**